velox/functions/lib/SIMDComparisionUtil.h

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/expression/VectorFunction.h"
+
+namespace facebook::velox::functions {
+
+namespace {
+
+template <typename T, bool isConstant, typename Arch = xsimd::default_arch>
+inline auto loadSimdData(const T* rawData, vector_size_t offset) {
+  using d_type = xsimd::batch<T>;
+  if constexpr (isConstant) {
+    return xsimd::broadcast<T>(rawData[0]);
+  }
+  return d_type::load_unaligned(rawData + offset);
+}
+
+template <bool allSelected>
+inline void setBoolTypeResultVectorByWord(
+    const uint64_t* rows,
+    uint8_t* result,
+    int8_t* tempRes,
+    vector_size_t index) {
+  if constexpr (allSelected) {
+    *(reinterpret_cast<uint32_t*>(result + index / 8)) = simd::toBitMask(
```

Comment:
You are assuming AVX2 with 32 bools in a register, which is not always the case.  It's better to fix the result width (64 bits) and have compile time dispatch on the SIMD register width for this function.

Reply:
Updated, thanks.